src/scripts/Controller.js

```diff
@@ -725,11 +734,44 @@ export default class SitemapController {
 				sitemapObj.model,
 				sitemapObj.selectors
 			);
+
+			if (sitemap.extensionVersion < 1) {
+				this.transformOldToNew(sitemap);
+			}
 			sitemap = await this.store.createSitemap(sitemap);
-			this._editSitemap(sitemap, ['_root']);
+			this._editSitemap(sitemap);
 		}
 	}
 
+	// transform old type sitemaps(without uuid) to new type sitemaps(with uuid)
+	transformOldToNew(sitemap) {
+		let parentsDict = { [sitemap.rootSelector.id]: sitemap.rootSelector.uuid };
+		let lastSelectorsUUID = 0;
+
+		sitemap.selectors.forEach(selector => {
+			if (selector.canHaveChildSelectors()) {
+				lastSelectorsUUID += 1;
+				parentsDict[selector.id] = lastSelectorsUUID.toString();
+			}
+		});
+		sitemap.selectors.forEach(selector => {
+			lastSelectorsUUID += 1;
+			selector.uuid =
+				parentsDict[selector.id] !== undefined
+					? parentsDict[selector.id]
+					: lastSelectorsUUID.toString();
+
+			const parents = Object.keys(parentsDict);
+
+			selector.parentSelectors.forEach((parentSelector, pSelectorIndex) => {
+				if (parents.indexOf(parentSelector) !== -1) {
+					selector.parentSelectors[pSelectorIndex] = parentsDict[parentSelector];
+				}
+			});
+		});
```

Comment:
Your approach is ok, but I think it would be more intuitive to assign uids to all selectors (not just possible parents) in the first pass and convert all parent selectors lists in the seconds pass. This way newly converted sitemap will have its selectors numbered exactly in their order.